routing/dht/routing.go

```diff
@@ -84,7 +84,10 @@ func (dht *IpfsDHT) GetValue(ctx context.Context, key key.Key) ([]byte, error) {
 	ctx, cancel := context.WithTimeout(ctx, time.Minute)
 	defer cancel()
 
-	vals, err := dht.GetValues(ctx, key, 16)
+	// retrieve a majority of the expected record count
+	majority := (KValue / 2) + 1
```

Comment:
Alright, so heres the part I disagree with. I dont think the number of records we require should be greater than the number of records that a single store operation puts. That is the reason for a lot of the slowness, we put out 10 records, and then try to get 16 from the network.